GitCommands/Git/GitBranchNameNormaliser.cs

```diff
@@ -107,8 +107,7 @@ public string Normalise(string branchName, GitBranchNameOptions options)
         /// </summary>
         public static bool IsValidChar(char c)
         {
-            return (c >= 32 && c < 127) &&
-                    c != ' ' && c != '~' && c != '^' && c != ':' &&
+            return (c is > ' ' and < '~' and not '^' and not ':') &&
```

Comment:
💭 No way I would have known that `'~'` is 0x7E. We should document why this range has been chosen.

Reply:
This is documented in the top of the file. I updated the link and compared to current (adding one rule).